proposals/p1382.md

```diff
@@ -0,0 +1,108 @@
+# Rename `me` -> `self`
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1382)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Don't change anything](#dont-change-anything)
+    -   [`this`](#this)
+
+<!-- tocstop -->
+
+## Problem
+
+We've tried the `fn MethodName[me: Self]()` syntax for a while, and from our
+experience the brevity of `me` is not worth doing something novel in this space.
+We have found that `me` doesn't read well in practice.
+
+## Background
+
+The current method syntax, including these choices, was decided in
+questions-for-leads issue
+[#494: Method syntax](https://github.com/carbon-language/carbon-lang/issues/494)
+and implemented in proposal
+[#722: Nominal classes and methods](https://github.com/carbon-language/carbon-lang/pull/722).
+
+Looking at other languages that use reserved word for the receiver value:
+
+| When       | Language   | Receiver when<br />accessing members | Receiver value | Receiver type |
+| ---------- | ---------- | ------------------------------------ | -------------- | ------------- |
+| 1983       | C++        | implicit                             | `this`         | ---           |
```

Comment:
I think it'd be good to add the explicit case of C++ as that is coming: https://wg21.link/p0847

Notably, while its *marking* the receiver with the `this` keyword, it names it `self` very consistently, and often even uses `Self` for the type in a similar way to what we're suggesting.

Reply:
Does https://github.com/carbon-language/carbon-lang/pull/1382/commits/57cd781a9aa8f081318c4528255779f4c543275c address this?